plugin/trino-geospatial/src/main/java/io/trino/plugin/geospatial/GeoFunctions.java

```diff
@@ -1538,18 +1540,17 @@ private static OGCGeometry geometryFromText(Slice input)
         return geometry;
     }
 
-    private static OGCGeometry geomFromBinary(Slice input)
+    private static Geometry geomFromBinary(Slice input)
     {
         requireNonNull(input, "input is null");
-        OGCGeometry geometry;
         try {
-            geometry = OGCGeometry.fromBinary(input.toByteBuffer().slice());
+            Geometry geometry = new WKBReader().read(input.getBytes());
+            geometry.setSRID(0);
```

Comment:
I think that this call is redundant per:

```
private Geometry read(InStream is, int maxCoordNum)
  throws IOException, ParseException
  {
    /**
     * This puts an upper bound on the allowed value
     * in coordNum fields.
     * It avoids OOM exceptions due to malformed input.
     */
    this.maxNumFieldValue = maxCoordNum;
    dis.setInStream(is);
    return readGeometry(0);
  }
```

Reply:
Thanks for the review!

The readGeomtry(0) might be a bit misleading. Later in the code the SRID is set for EWKB geometries. https://github.com/locationtech/jts/blob/master/modules/core/src/main/java/org/locationtech/jts/io/WKBReader.java#L260.

But it might be redundant anyway since SRID is discarded in the serializer. Should I remove the serSRID(0) call?

Reply:
Please do